tests/experimental/datasources/test_postgres_datasource.py

```diff
@@ -700,3 +711,133 @@ def test_validate_invalid_postgres_connection_string(
             connection_string=connection_string,
         ):
             pass
+
+
+def bad_connection_string_config() -> tuple[
+    str, str, str, str | None, TestConnectionError
+]:
+    connection_string = "postgresql+psycopg2://postgres:@localhost/bad_database"
+    table_name = "good_table"
+    schema_name = "good_schema"
+    sqlalchemy_engine_connect_error_raised = (
```

Comment:
I find testing exact strings to be fragile. I'm fine with checking the error type. If you want to check the string maybe look for the substring `database "bad_database" does not exist`? Likely this won't change but it's not part of any public api.

Reply:
This is mocked so no external dependency changes will cause this to fail. The test just ensures that this string is passed into the `TestConnectionError`.

Reply:
I changed the variable name to `mock_sqlalchemy_engine_connect_error_raised`.